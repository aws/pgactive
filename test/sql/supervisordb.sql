\c postgres

-- The DB name pgactive_supervisordb is reserved by pgactive. None
-- of these commands may be permitted.

CREATE DATABASE pgactive_supervisordb;

DROP DATABASE pgactive_supervisordb;

ALTER DATABASE pgactive_supervisordb RENAME TO someothername;

ALTER DATABASE regression RENAME TO pgactive_supervisordb;

-- We can connect to the supervisor db...
\c pgactive_supervisordb

SET log_statement = 'all';

-- We actually did connect
SELECT current_database();

-- And do read-only work
SELECT 1;

-- but not do anything interesting
CREATE TABLE create_fails(id integer);

\d

-- except vacuum
VACUUM;
