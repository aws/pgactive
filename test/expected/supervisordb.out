\c postgres
-- The DB name pgactive_supervisordb is reserved by pgactive. None
-- of these commands may be permitted.
CREATE DATABASE pgactive_supervisordb;
ERROR:  pgactive extension reserves the database name pgactive_supervisordb for its own use
HINT:  Use a different database name.
DROP DATABASE pgactive_supervisordb;
ERROR:  pgactive extension reserves the database name pgactive_supervisordb for its own use
HINT:  Use a different database name.
ALTER DATABASE pgactive_supervisordb RENAME TO someothername;
ERROR:  pgactive extension reserves the database name pgactive_supervisordb for its own use
HINT:  Use a different database name.
ALTER DATABASE regression RENAME TO pgactive_supervisordb;
ERROR:  pgactive extension reserves the database name pgactive_supervisordb for its own use
HINT:  Use a different database name.
-- We can connect to the supervisor db...
\c pgactive_supervisordb
SET log_statement = 'all';
ERROR:  no commands may be run on the pgactive supervisor database
-- We actually did connect
SELECT current_database();
 current_database 
------------------
 pgactive_supervisordb
(1 row)

-- And do read-only work
SELECT 1;
 ?column? 
----------
        1
(1 row)

-- but not do anything interesting
CREATE TABLE create_fails(id integer);
ERROR:  no commands may be run on the pgactive supervisor database
\d
      List of relations
 Schema | Name | Type | Owner 
--------+------+------+-------
(0 rows)

-- except vacuum
VACUUM;
